src/Build/BuildCheck/Analyzers/NoEnvironmentVariablePropertyAnalyzer.cs

```diff
@@ -9,25 +9,39 @@ namespace Microsoft.Build.Experimental.BuildCheck.Analyzers;
 
 internal sealed class NoEnvironmentVariablePropertyAnalyzer : BuildAnalyzer
 {
+    public static BuildAnalyzerRule SupportedRule = new BuildAnalyzerRule(
+                "BC0103",
+                "NoEnvironmentVariablePropertyAnalyzer",
+                "No implicit property derived from an environment variable should be used during the build",
+                StandardMessage,
+                new BuildAnalyzerConfiguration() { Severity = BuildAnalyzerResultSeverity.Suggestion });
+
+    private const string RuleId = "BC0103";
+
+    private const string VerboseEnvVariableOutputKey = "allow_displaying_environment_variable_value";
+
     /// <summary>
     /// Contains the list of reported environment variables.
     /// </summary>
     private readonly HashSet<EnvironmentVariableIdentityKey> _environmentVariablesReported = new HashSet<EnvironmentVariableIdentityKey>();
 
-    public static BuildAnalyzerRule SupportedRule = new BuildAnalyzerRule(
-        "BC0103",
-        "NoEnvironmentVariablePropertyAnalyzer",
-        "No implicit property derived from an environment variable should be used during the build",
-        "Property is derived from environment variable: '{0}' with value: '{1}'. Properties should be passed explicitly using the /p option.",
-        new BuildAnalyzerConfiguration() { Severity = BuildAnalyzerResultSeverity.Suggestion });
+    private const string VerboseMessage = "Property is derived from environment variable: '{0}' with value: '{1}'. Properties should be passed explicitly using the /p option.";
+
+    private const string StandardMessage = "Property is derived from environment variable: '{0}'. Properties should be passed explicitly using the /p option.";
+
+    private bool _isVerboseEnvVarOutput;
 
     public override string FriendlyName => "MSBuild.NoEnvironmentVariablePropertyAnalyzer";
 
     public override IReadOnlyList<BuildAnalyzerRule> SupportedRules { get; } = [SupportedRule];
 
     public override void Initialize(ConfigurationContext configurationContext)
     {
-        // No custom configuration
+        foreach (CustomConfigurationData customConfigurationData in configurationContext.CustomConfigurationData)
+        {
+            bool? isVerboseEnvVarOutput = GetVerboseEnvVarOutputConfig(customConfigurationData, RuleId);
+            _isVerboseEnvVarOutput = isVerboseEnvVarOutput.HasValue && isVerboseEnvVarOutput.Value;           
+        }
```

Comment:
Your analyzer has just single rule - so you can as well call just `GetVerboseEnvVarOutputConfig(configurationContext.CustomConfigurationData[0], RuleId);`

Reply:
But your code is fine as well and actually more universal - so let's maybe keep it :-)